sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -104,10 +104,23 @@ public static QueryNode translateQuery(Map<String, Object> query,
         TranslatorState state = new TranslatorState();
         QueryNode node = translateQuery(query, indexes, state);
 
-        // If we haven't used a single index or an OR clause is missing an index,
-        // we need to return a query which returns every document, so the posthoc
-        // matcher can run over every document to manually carry out the query.
-        if (!state.atLeastOneIndexUsed || state.atLeastOneORIndexMissing) {
+        if (state.textIndexMissing) {
+            String msg = "No text index defined, cannot execute query containing a text search.";
+            logger.log(Level.SEVERE, msg);
+            return null;
+        } else if (state.textIndexRequired && state.atLeastOneIndexMissing) {
+            String msg = String.format("Query %s contains a text search but is missing \"json\"" +
+                                       " index(es).  All indexes must exist in order to execute a" +
+                                       " query containing a text search.  Create all necessary" +
+                                       " indexes for the query and re-execute.",
+                                       query.toString());
+            logger.log(Level.SEVERE, msg);
```

Comment:
Before FTS we would pass any query that had at least one missing index to the post hoc matcher where we would then take the full list of doc ids in a datastore and inspect each document and find query matches programmatically.  I think early on we agreed to limit this FTS initial implementation to the SQL engine only since implementing a post hoc matcher component would be non-trivial.  That being the case we must for now "turn off" matcher functionality for any query that contains a FTS predicate and mandate that it be handled by the SQL engine only.  So I guess to answer your question, is that it does not appear to be anything fundamental that dictates this, but instead was a design choice made for this initial implementation.